src/rans/rANSCoder.py

```diff
@@ -50,10 +56,15 @@ def find_in_int_dist(cdf, to_find):
 
     print ("Error: Could not find symbol in integer-dist")
 
+spec = [
+    ('state', numba.int64),
+    ('encoded_data', types.ListType(types.int64)),
```

Comment:
If I recall correctly, `encoded_data` is actually a list of 32-bit ints - and state is a 64-bit int. Could you specify `ListType(types.int32)` here for `encoded_data`?

Reply:
hmm looks like you're right but I'm having some trouble getting encoding then decoding to work properly w/         `encoded_data` as a list of `types.int32`; the bug seems as if it appears that there are too few elements in the list so it then calls .pop on an empty list?

```
  0%|          | 0/10000 [00:00<?, ?it/s]<string>:3: NumbaTypeSafetyWarning: unsafe cast from int64 to int32. Precision may be lost.
100%|██████████| 10000/10000 [00:00<00:00, 19817.50it/s]
 30%|██▉       | 2965/10000 [00:00<00:00, 10671.14it/s]
Traceback (most recent call last):
  File "main.py", line 20, in <module>
    decoded[i] = d.decode_symbol(prb)
  File "venv\lib\site-packages\numba\experimental\jitclass\boxing.py", line 61, in wrapper
    return method(*args, **kwargs)
  File "venv\lib\site-packages\numba\typed\listobject.py", line 922, in impl
    raise IndexError("pop from empty list")
IndexError: pop from empty list

```

This is my script btw

```
import rans.rANSCoder as rANS
import numpy as np
from tqdm import tqdm


if __name__ == '__main__':
    with open('enwik8', 'rb') as f:
        data = np.array(bytearray(f.read()[:10000]), dtype=np.uint8)

    e = rANS.Encoder()
    uniq, idx, cts = np.unique(data, return_counts=True, return_index=True)
    prb = np.zeros((256,), dtype=np.float32)
    prb[uniq] = cts / np.sum(cts)
    for sym in tqdm(data):
        e.encode_symbol(prb, sym)
    encoded = e.get_encoded()
    d = rANS.Decoder(encoded)
    decoded = np.zeros((len(data),), dtype=np.uint8)
    for i in tqdm(range(len(data))):
        decoded[i] = d.decode_symbol(prb)
    print(np.all(data == decoded[::-1]))
```

Reply:
hmm—maybe try `types.uint32`? The C++ LIbrary I ported the code from uses `uint32_t`, there might be some edge cases there where the first bit needs to be interpreted as part of the number and not the sign.

if that doesn't work i'll try to take a deeper look on this weekend.

Reply:
That worked! And it's a bit faster now too

Reply:
Excellent! Then I'll play around your code on the weekend a bit and 99% chance I'll merge it in afterwards :smile: 

Thanks a lot!